src/components/datepicker/datePicker.js

```diff
@@ -286,19 +387,55 @@
   DatePickerCtrl.prototype.attachChangeListeners = function() {
     var self = this;
 
-    self.$scope.$on('md-calendar-change', function(event, date) {
-      self.ngModelCtrl.$setViewValue(date);
-      self.date = date;
-      self.inputElement.value = self.dateLocale.formatDate(date);
-      self.closeCalendarPane();
-      self.resizeInputElement();
-      self.updateErrorState();
-    });
-
-    self.ngInputElement.on('input', angular.bind(self, self.resizeInputElement));
-    // TODO(chenmike): Add ability for users to specify this interval.
-    self.ngInputElement.on('input', self.$mdUtil.debounce(self.handleInputEvent,
+    // Cannot use isDateRangePicker flag here as this even listener is wired up before the flag gets assigned a value.
+    if (self.$scope.ctrl.begin == undefined && self.$scope.ctrl.end == undefined) {
```

Comment:
As you mentioned, this check is no longer needed given the impl of your prior mentioned suggestion around initially setting the isDatePicker value.